python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -370,12 +376,31 @@ def materialize_on_required_for_freshness() -> "MaterializeOnRequiredForFreshnes
 
     @public
     @staticmethod
-    def materialize_on_parent_updated() -> "MaterializeOnParentUpdatedRule":
-        """Materialize an asset partition if one of its parents has been updated more recently
-        than it has.
+    def materialize_on_cron(
+        cron_schedule: str, timezone: str = "UTC", all_partitions: bool = False
+    ) -> "MaterializeOnCronRule":
+        """Materialize an asset partition if it has not been materialized since the previous cron
+        schedule tick.
 
-        Note: For time-partitioned or dynamic-partitioned assets downstream of an unpartitioned
-        asset, this rule will only fire for the most recent partition of the downstream.
+        Args:
+            cron_schedule (str): A cron schedule string (e.g. "0 * * * *") indicating the ticks for
+                which this rule should fire.
+            timezone (str): The timezone in which this cron schedule should be evaluated. Defaults
+                to "UTC".
+            all_partitions (bool): If True, this rule fires for all partitions of this asset on each
```

Comment:
Should we support catchup? I.e. suppose that the scheduler falls behind because of ticks taking a long time or something or some skip rule – it might be useful for the user not to need to manually monitor to make sure there are no partition gaps.

I think we support this to some degrees with regular schedules / `build_schedule_from_partitioned_job`, though would have to dig to find the exact semantics.

Reply:
yeah agree here -- the semantics of catching up in schedule land are handled entirely by the ScheduleDaemon itself (it runs the schedule code independently on every tick it missed, up to max_catchup_runs)

could definitely do something similar there, and maybe we don't need to allow the user to set an analog to "max_catchup_runs" at all, as that could be controlled by the "max_materializations_per_minute" parameter

Reply:
updated this so that it will now "catch up"